src/block/parser.rs

```diff
@@ -234,11 +242,13 @@ impl BlockParser {
                 canonical_paths.reverse();
 
                 // add all blocks successive to the canonical chain
-                for path in paths[successive_idx..]
-                    .iter()
-                    .filter(|p| length_from_path(p).is_some())
-                {
-                    successive_paths.push(path.clone());
+                if let Some(successive_idx) = successive_idx {
```

Comment:
maybe too much of a nitpick but `successive_idx.map(|idx| {...})` would be more idiomatic here